### PR TITLE
List all python distribution extras in tox env declaration

### LINKED ISSUE #1689: tox.ini should reference all extras present in setup.cfg
# Issue Type

- Bug report

# Actual Behaviour

Some extras present in setup.cfg are missing in tox.ini and should be added.

In setup.cfg:

```
[extras]
docs =
    alabaster
    Sphinx
azure =
    ansible[azure]
docker =
    docker>=2.0.0
ec2 =
    boto
    boto3
gce =
    apache-libcloud
linode =
    linode-python; python_version<"3.0"
lxc =
    lxc-python2; python_version<"3.0"
openstack =
    shade
vagrant =
    python-vagrant
windows =
    pywinrm
```

in tox.ini:

```
extras =
    azure
    docker
    gce
    lxc
    openstack
    vagrant
```

### PR DESCRIPTION
**Pull request by themr0c**
Signed-off-by: Fabrice Flore-Thebault <themr0c@users.noreply.github.com>

Closes #1689

#### PR Type

- Bugfix Pull Request

### CODE REVIEW

File: tox.ini [hunk 1]
```diff
@@ -24,10 +24,14 @@ deps =
...
     azure
     docker
+    docs
+    ec2
     gce
+    linode
     lxc
     openstack
     vagrant
+    windows
```
Comment (webknjaz) [inline tox.ini]: Is this conditional in `setup.cfg`

**Comment by webknjaz** [comment]
Also linters fail

File: tox.ini (hunk 1 above)
Comment (themr0c) [reply tox.ini]: I don't understand ?

**Comment by themr0c** [comment]
@webknjaz You are talking about the linting issue there,  https://travis-ci.com/ansible/molecule/jobs/172385408 : all builds are failing on lint and this PR is the attempt to fix the lint issue https://github.com/ansible/molecule/pull/1695

**Comment by webknjaz** [comment]
I'd probably want to fix linting issues in a separate PR so that those changes wouldn't pollute other PRs

File: tox.ini (hunk 1 above)
Comment (webknjaz) [reply tox.ini]: I'm curious whether it will try installing incompatible dep under Linux and then running tests against it

Comment (themr0c) [reply tox.ini]: Ah ! certainly not, the windows extra is to install pywinrm, the module to be able to connect to windows from ansible.

**Comment by lwm** [comment]
Can we merge this?

**Comment by themr0c** [comment]
Sorry i failed a rebase again, hence the push --force. But it is just the exact same commit rebased on master to relaunch the build which failed on a weird error.

**Comment by themr0c** [comment]
Unfortunately, with this PR the  `linode` modules get installed, linode tests that were skipped get activated, and they fail.

**Comment by lwm** [comment]
Riiighhhhht, OK, we can't merge this until we have a strategy for these functional tests that talk across the network.

**Comment by webknjaz** [comment]
@themr0c right, that's what I thought...
You can modify https://github.com/ansible/molecule/blob/6502649/test/functional/conftest.py#L268-L274 to ignore travis as a workaround.

**Comment by themr0c** [comment]
If i didn't make any enormous mistake in last commit, linode functional tests should be now skipped in travis.

File: test/functional/conftest.py [hunk 2]
```diff
@@ -268,6 +268,14 @@ def get_docker_executable():
     return distutils.spawn.find_executable('docker')
 
 
+def get_linode_executable():
+    try:
+        pytest.importorskip('linode')
+        return True
+    except Exception:
```
Comment (webknjaz) [inline test/functional/conftest.py]: What exception is actually happening here? AFAIK `pytest.importorskip` with just return `None` if there's nothing to import. Which means that this check'd always return `True`.

`pytest.importorskip` is specifically designed to trigger skipping the current test anyway so I don't know why you would wrap it like this.

Comment (webknjaz) [reply test/functional/conftest.py]: Also, a bunch of unrelated changes makes it difficult to guess what is the purpose of the PR...

Comment (webknjaz) [reply test/functional/conftest.py]: Oh.. I see now.

File: test/functional/conftest.py [hunk 3]
```diff
@@ -291,11 +299,8 @@ def supports_docker():
 
 @pytest.helpers.register
 def supports_linode():
-    try:
-        pytest.importorskip('linode')
-        return True
-    except Exception:
-        return False
+    # FIXME: Travis CI
+    return not IS_TRAVIS and get_lxd_executable()
```
Comment (lwm) [inline test/functional/conftest.py]: `%s/get_lxd_executable/get_linode_executable/`

File: test/functional/conftest.py (hunk 2 above)
Comment (webknjaz) [reply test/functional/conftest.py]: I'm not sure how this check works but it's probably not related to the scope of the current PR. So let's ignore it for now.

File: test/functional/conftest.py (hunk 3 above)
Comment (webknjaz) [reply test/functional/conftest.py]: oh... I missed that. btw you can suggest changes in PRs with `Ctrl+G`:
```suggestion
    return not IS_TRAVIS and get_linode_executable()
```

**Comment by lwm** [comment]
@themr0c, I think https://github.com/ansible/molecule/pull/1691#discussion_r258524200 is relevant in that, `importorskip` won't actually throw an exception. Let's make sure about that.

**Comment by webknjaz** [comment]
@lwm while it's relevant to the initial code, it's just copy-pasted without changes so I suggest to address it separately in order to not overload PR with extra changes

File: test/functional/conftest.py (hunk 2 above)
Comment (themr0c) [reply test/functional/conftest.py]: I just wanted to align on the other platforms where there is a split between def_xxx_executable and a supports_xxx.
The try and exception is just a copy paste during the split, nothing new here.

Comment (webknjaz) [reply test/functional/conftest.py]: Yes, I noticed that already and suggested to address it separately in other comments.

**Comment by lwm** [comment]
:rocket:

**Pull request merged**